pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PublicSettings.scala

```diff
@@ -1,88 +1,56 @@
+import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
 
-import akka.agent.Agent
-import com.gu.pandomainauth.PublicSettings.FunctionJob
-import okhttp3._
-import org.quartz._
-import org.quartz.impl.StdSchedulerFactory
+import com.amazonaws.services.s3.AmazonS3
+import org.slf4j.LoggerFactory
 
-import scala.collection.JavaConverters._
-import scala.collection.mutable
-import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
 
 /**
- * Class that contains the static public settings and includes mechanism for fetching the public key. The class is
- * parametrized by domain because the keys will differ for different domains. It also requires a dispatch.Http
- * instance for retrieving the data and an execution context.
- *
- * Once you have an instance you can call its start method to kick off loading the public data. In a Play app,
- * consider using the `Global` object's `onStart` method to start these calls when the application comes up. If
- * you'd rather use your own scheduler you can do so while still using the PublicSettings class by scheduling
- * your own calls to its `refresh` method can also schdule the refresh yourself using the instance's
- *
- * @param domain     The domain you would like to fetch settings for
- * @param callback   Optionally, a callback called when the data gets fetched, provided to allow you to perform logging
- * @param bucketName The name of the S3 bucket where the settings are stored
- * @param scheduler  Optionally, the quartz scheduler instance to use. It defaults to the default scheduler but
- *                   customising it may be useful if you want more control over the scheduler's lifecycle
- * @param client     Implicit instance of dispatch.Http used to make the call to fetch the public settings
- * @param ec         Implicit execution context used to fetch the settings
+ * Class that contains the static public settings and includes mechanism for fetching the public key. Once you have an
+ * instance, call the `start()` method to load the public data.
+  *
+  * @param settingsFileKey the settings file for the domain in the S3 bucket (eg local.dev.gutools.co.uk.public.settings)
+  * @param bucketName      the name of the S3 bucket (eg pan-domain-auth-settings)
+  * @param s3Client        the AWS S3 client that will be used to download the settings from the bucket
+  * @param scheduler       optional scheduler that will be used to run the code that updates the bucket
  */
-class PublicSettings(domain: String, bucketName: String, callback: Try[Map[String, String]] => Unit = _ => (),
-                     scheduler: Scheduler = StdSchedulerFactory.getDefaultScheduler())
-                    (implicit client: OkHttpClient, ec: ExecutionContext) {
+class PublicSettings(settingsFileKey: String, bucketName: String, s3Client: AmazonS3,
```

Comment:
`PublicSettings` previously made an HTTP request to retrieve the settings file. This added *another* dependency (OkHttp) and as far as I could see provided no value over using the normal S3 client other than the request being asynchronous. The only piece of code that [reads the public settings directly](https://github.com/guardian/content-api/blob/99cf94c31ae3b3dba4618b30cce6ac90c636286a/gateways/panda-lambda/src/main/scala/com/gu/pandalambda/Lambda.scala#L30) added an `Await` around it anyway!

I think it's clearer and more consistent if we use an S3 client in this class.